### PR TITLE
PR #45: shooter w/o pivot (dev to main)

### PR DESCRIPTION
**Pull request by Irishumanoid**

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -14,43 +10,66 @@
...
 
-    m_intake = new Intake();
-
-    configureBindings();
     registerCommands();
+    // adds all autos in deploy dir to chooser
+    autoChooser = AutoBuilder.buildAutoChooser();
+    configureBindings();
+
+    /*m_shooter.setDefaultCommand(
```
Comment (cdvdmackey) [inline src/main/java/frc/robot/RobotContainer.java:65]: remove this comment

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -67,6 +86,33 @@ public RobotContainer() {
...
+
+    // triggers for extending and retracting shield manually
+    // don't extend shield
+    new Trigger(() -> m_operatorController.getRawButton(Bindings.kExtendShield))
+        .onTrue(new ActuateShield(m_shooter, false));
+    // extend shield
+    new Trigger(() -> m_operatorController.getRawButton(Bindings.kRetractShield))
+        .onTrue(new ActuateShield(m_shooter, true));
+
+    autoChooser.setDefaultOption("Leave Top", AutoBuilder.buildAuto("LeaveFromTop"));
```
Comment (cdvdmackey) [inline src/main/java/frc/robot/RobotContainer.java:115]: what does this auto do? is it tested.

File: src/main/java/frc/robot/constants/RobotConfig.java [hunk 3]
```diff
@@ -3,14 +3,130 @@
...
+      public static final double maxIntegral = 2;
+    }
+
+    public static class TurnConfig {
+      public static final String kPKey = "Vision Turn P";
+      public static final String kIKey = "Vision Turn I";
+      public static final String kDKey = "Vision Turn D";
+      public static final double kP = 0.0;
+      public static final double kI = 0.0;
+      public static final double kD = 0.0;
```
Comment (awfulbananas) [inline src/main/java/frc/robot/constants/RobotConfig.java:124]: Should the PID's have non-0 values?

File: src/main/java/frc/robot/subsystems/intake/Intake.java [hunk 4]
```diff
@@ -8,12 +8,12 @@
 import frc.robot.constants.RobotConstants.IntakeConstants;
 
 public class Intake extends SubsystemBase {
-  private final CANSparkMax m_rollerMotor; // Intake roller motor
+  private final CANSparkMax m_intakeRollerMotor; // Intake roller motor
   private final DigitalInput m_linebreak;
 
-  // Constructs intake and initializes motor
+  /** Creates a new ExampleSubsystem. */
```
Comment (awfulbananas) [inline src/main/java/frc/robot/subsystems/intake/Intake.java:14]: Probably best to remove this comment

File: src/main/java/frc/robot/commands/shooter/Aim.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    } else {
+      switch (m_type) {
+        case AMP:
+          desiredAngle = ShooterConfig.kAmpAngle;
+          desiredVelocity = ShooterConfig.ampVelocity;
+          break;
+        case SPEAKER:
+          desiredAngle = ShooterConfig.kSpeakerAngle;
+          desiredVelocity = ShooterConfig.speakerVelocity;
+          // getVelocity(ShooterConfig.SpeakerHeight);
```
Comment (cdvdmackey) [inline src/main/java/frc/robot/commands/shooter/Aim.java:58]: more commented stuff we should remove this when possible

**Review by awfulbananas — APPROVED** [review]
As long as those pid values are meant to be 0 this looks good to me

File: src/main/java/frc/robot/constants/RobotConfig.java [hunk 6]
```diff
@@ -3,14 +3,130 @@
...
+
+    public static final double SpeakerBillLength = 0.6604;
+
+    public static final int shootButton = 1;
+
+    public static final double kDefaultFlywheelRPM = 1000;
+
+    public static final double kShooterStowAngle = 0;
+
+    public static final long kReleaseTime = 5000;
```
Comment (cdvdmackey) [inline src/main/java/frc/robot/constants/RobotConfig.java:89]: units, pretty obv its in ms but not the best

File: src/main/java/frc/robot/constants/RobotConfig.java [hunk 7]
```diff
@@ -3,14 +3,130 @@
...
+    public static final long kShieldTime = 2; // seconds
+    public static final double kShieldDefaultSpeed = 0.5;
+    public static final Measure<Velocity<Angle>> kFlywheelError = Units.RPM.of(1);
+    public static final Measure<Angle> kAngleError = Units.Radians.of(0.5 * Math.PI / 180);
+    public static final Measure<Angle> kSpeakerAngle = Units.Radians.of(75 * Math.PI / 180);
+    public static final Measure<Angle> kAmpAngle = Units.Radians.of(109 * Math.PI / 180);
+    public static final Measure<Angle> kTrapAngle = Units.Radians.of(105 * Math.PI / 180);
+    public static final Measure<Angle> kAdjustAmountDegrees = Units.Radians.of(0.5 * Math.PI / 180);
+
+    // TODO placeholders
```
Comment (cdvdmackey) [inline src/main/java/frc/robot/constants/RobotConfig.java:99]: why are these so low and are they still placeholders and are they being used in the flywheel speed

File: src/main/java/frc/robot/constants/RobotConstants.java [hunk 8]
```diff
@@ -1,19 +1,67 @@
...
+    public static final int kExtendShield = 9;
+    public static final int kManualAdjustDown = 18;
+    public static final int kManualAdjustUp = 19;
+  }
+
+  public static final class VisionConstants {
+    public static final double camChassisXOffset = -1;
+    public static final double camChassisYOffset = -1;
+    public static final double camChassisZOffset = -1;
+
```
Comment (cdvdmackey) [inline src/main/java/frc/robot/constants/RobotConstants.java:39]: why do we have vision files if the constants for them are not even set? if were going to implement during comp we should have these set or written down somewhere

File: src/main/java/frc/robot/constants/RobotConstants.java [hunk 9]
```diff
@@ -1,19 +1,67 @@
...
     public static final double kFreeSpeedRpm = 5676;
   }
 
+  public final class ShooterConstants {
+    public static final int kRollerMotorLeftId = 15;
+    public static final int kTopFlywheelMotorId = 16;
+    public static final int kBottomFlywheelMotorId = 17;
+    public static final int kShieldMotorId = 18;
+    public static final double FlywheelDiameter = 0.0762;
+    public static final double ShooterLength = 0.4064;
```
Comment (cdvdmackey) [inline src/main/java/frc/robot/constants/RobotConstants.java:59]: arent using this constant anymore

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 10]
```diff
@@ -1,18 +1,212 @@
...
+    SmartDashboard.putNumber("flywheel d", m_topFlywheelPIDController.getD());
+
+    m_targetVelocity = MutableMeasure.zero(Units.MetersPerSecond);
+
+    if (DriverStation.isTest()) {
+      putAngleOnSmartDashboard();
+    }
+  }
+
+  public void putAngleOnSmartDashboard() {
```
Comment (cdvdmackey) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java:82]: confusing method name

File: src/main/java/frc/robot/subsystems/shooter/Shooter.java [hunk 11]
```diff
@@ -1,18 +1,212 @@
...
+
+  public double convertToRPM(double velocity) {
+    // 0.0762 meters is diameter of flywheel
+    double circumference = ShooterConstants.FlywheelDiameter * Math.PI * 60;
+    double rpm = velocity / circumference;
+    return rpm;
+  }
+
+  // returns true if extended
+  public boolean getShieldStatus(boolean extend) {
```
Comment (cdvdmackey) [inline src/main/java/frc/robot/subsystems/shooter/Shooter.java:192]: why do we still have both `getShieldStatus()` and `getShieldPosition()`

**Review by cdvdmackey — APPROVED** [review]
overall it looks fine, but having unused vision code with unset constants is not good practice even though there are no triggers for it

File: src/main/java/frc/robot/commands/VisionTranslateCommand.java [hunk 12]
```diff
@@ -0,0 +1,67 @@
+package frc.robot.commands;
```
Comment (ChainedGCH) [inline src/main/java/frc/robot/commands/VisionTranslateCommand.java:1]: Do we not have a vision commands folder?

File: src/main/java/frc/robot/subsystems/vision/Vision.java [hunk 13]
```diff
@@ -1,18 +1,110 @@
...
+
+  private PhotonCamera camera;
+  private boolean hasTarget;
+  private PhotonPipelineResult result;
+
+  private AprilTagFieldLayout aprilTagFieldLayout;
+  // if there's a pose estimator in the drivetrain subsystem, update it with this estimator
+  private PhotonPoseEstimator poseEstimator;
+
+  public Vision() {
```
Comment (TurtleMeds) [inline src/main/java/frc/robot/subsystems/vision/Vision.java:30]: Why are we messing with vision code right now if we aren't using it?

**Review by TurtleMeds — APPROVED** [review]

File: src/main/java/frc/robot/constants/RobotConfig.java [hunk 14]
```diff
@@ -3,14 +3,130 @@
...
+    public static final double kAngleControlP = 0;
+    public static final double kAngleControlI = 0;
+    public static final double kAngleControlD = 0;
+    public static final double kAngleControlFF = 0;
+    public static final double kAngleControlIZone = 0;
+    public static final double kAngleControlMinOutput = 0;
+    public static final double kAngleControlMaxOutput = 0;
+
+    // top Flywheel controller PID coefficients
+    public static final double kTopFlywheelP = 0.2;
```
Comment (ChainedGCH) [inline src/main/java/frc/robot/constants/RobotConfig.java:40]: Are these values tuned?

File: src/main/java/frc/robot/constants/RobotConfig.java (hunk 3 above)
Comment (cdvdmackey) [reply src/main/java/frc/robot/constants/RobotConfig.java:124]: yep

Comment (continuumsrc) [reply src/main/java/frc/robot/constants/RobotConfig.java:124]: Anything related to vision is unused so 0 pid constants on those is probably fine.

File: src/main/java/frc/robot/constants/RobotConfig.java (hunk 14 above)
Comment (Irishumanoid) [reply src/main/java/frc/robot/constants/RobotConfig.java:40]: yes

File: src/main/java/frc/robot/commands/VisionTranslateCommand.java (hunk 12 above)
Comment (Irishumanoid) [reply src/main/java/frc/robot/commands/VisionTranslateCommand.java:1]: not necessary, but could add later

File: src/main/java/frc/robot/RobotContainer.java (hunk 2 above)
Comment (Irishumanoid) [reply src/main/java/frc/robot/RobotContainer.java:115]: none are tested but this one just leaves

File: src/main/java/frc/robot/subsystems/intake/Intake.java (hunk 4 above)
Comment (cdvdmackey) [reply src/main/java/frc/robot/subsystems/intake/Intake.java:14]: yep we should but as long as its commented were keeping it

File: src/main/java/frc/robot/constants/RobotConfig.java (hunk 6 above)
Comment (cdvdmackey) [reply src/main/java/frc/robot/constants/RobotConfig.java:89]: It doesn't affect the functionality of the code so it should be fine right now

File: src/main/java/frc/robot/constants/RobotConfig.java (hunk 7 above)
Comment (cdvdmackey) [reply src/main/java/frc/robot/constants/RobotConfig.java:99]: These are the correct values. Although they are pretty weird.

**Pull request merged**